pkg/identity/numericidentity.go

```diff
@@ -287,10 +287,14 @@ func InitWellKnownIdentities(c Configuration) int {
 	//   k8s:io.kubernetes.pod.namespace=<NAMESPACE>
 	//   k8s:name=cilium-operator
 	//   k8s:io.cilium/app=operator
+	//   k8s:app.kubernetes.io/part-of=cilium
+	//   k8s:app.kubernetes.io/name=cilium-operator
 	//   k8s:io.cilium.k8s.policy.cluster=default
 	ciliumOperatorLabels := []string{
 		"k8s:name=cilium-operator",
 		"k8s:io.cilium/app=operator",
+		"k8s:app.kubernetes.io/part-of=cilium",
+		"k8s:app.kubernetes.io/name=cilium-operator",
```

Comment:
This unfortunately does not work. We want to keep the existing label selector as is, and create a new one for the extended labels.

You will have change the code in such a way that it supports both the old and the new label set.  Unfortunately, I just realized that we have to support both the versions with and without the `PodNamespaceMetaNameLabel`, because I believe we want to support K8s older than v1.21 still.

So in the end, there will have to be 4 calls to `WellKnown.add`, one for each label set:

```
WellKnown.add(ReservedCiliumOperator, ciliumOperatorLabels)
WellKnown.add(ReservedCiliumOperator2, ciliumOperatorLabelsWithNamespace)
WellKnown.add(ReservedCiliumOperator3, ciliumOperatorLabelsWithRecommendedLabels)
WellKnown.add(ReservedCiliumOperator4, ciliumOperatorLabelsWithRecommendedLabelsAndNamespace)
```

However, this will double the number of reserved identities related to the operator and etcd server. I'm not sure this is an approach that scale. Thoughts @aanm?

Reply:
That's a good question, but then I wonder: The label changes will part of Cilium version X.Y.Z and the well known identities will also be part of the version X.Y.Z, so why would we need to duplicate them?

Reply:
I guess there could be a time during the rollout where some agent pods still use the old label set, while the operator or etcd server has already been upgraded to the new labels. Especially for etcd, that might cause some disruption during the upgrade, because none of the old agents will be able to connect to the new etcd server until they have been upgraded too.

Reply:
It something that we would need to double check but I would guess that wouldn't be a problem because the security identity would end up being the same.

Reply:
So, if we're not worried about supporting the label set of versions N+1 or N-1, then I guess we can leave this as is.